addons/actions/src/preview/withActions.ts

```diff
@@ -47,8 +50,25 @@ export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn:
     }
     return undefined;
   }, [root, actionsFn, args]);
+}
 
+export const createDecorator = (actionsFn: any) => (...args: any[]) => (storyFn: () => any) => {
```

Comment:
How is this used? I think we should drop it and just allow deprecated usage below..

Reply:
It ends up getting used like this example:

https://github.com/storybookjs/storybook/blob/next/examples/html-kitchen-sink/stories/addon-actions.stories.js#L40

Feels weird to me but maybe  I'm not understanding the use case.

Reply:
Oh, wow that's pretty weird. I suspect this is a super old usage pattern from before we had support for story-level decorators, this was the only way to apply a decorator to a single story.

I'm not sure we need to keep supporting it. WDYT @shilman?

Reply:
Yeah I think we should drop support for this too.

Reply:
There isn't a great way to remove this without removing the ability to manipulate the event data for frameworks like html and web components. 

The problem is if you use parameters then the presetDecorator will apply but it would have the customer actionFunctions that `decorate` otherwise adds. We can't use parameters to supply the actionFunctions as they have to be serializable (so args v1 won't help here either). 

Is there some other mechanism where we can 'configure' the decorator to use a different set of action functions? @tmeasday @shilman  ?

Reply:
@jonspalmer I don't really understand what that use case is doing.

Reply:
@tmeasday I'm not too sure of it either. It looks like we went out of our way to allow people to manipulate the event arguments. I'm not clear why you want that.

In React land (or some other JS component framework) if you wanted to manipulate the args of the event handler you could do this without any special support from the actions addon:

```js
import { action } from '@storybook/addon-actions';
import Button from './button';

export default {
  title: 'Button',
  component: Button,
};

const clickAction = action('click');

export const first = () => (
  <Button onClick={(...args) => clickAction(args[0])}>Hello World!</Button>
);
```

However in html or web components we can't do it in the story function. So the equivalent is:

```js
import { decorate } from '@storybook/addon-actions';

const firstArg = decorate([args => [args[0]]]);

export const first = () => firstArg.withActions('click')(
  () => `<button type="button">Hello World!</button>`;
);
```

If its important to support the manipulate of event args I don't think we have a great way to maintain support other than this approach. However, I could see us dropping it entirely.

Reply:
After a conversation with @shilman we agreed the decorate API is YAGNI. I've removed it and updated the docs.

Reply:
Thanks for explaining it to me anyway @jonspalmer, I get it now ;)